src/core/wopi.py

```diff
@@ -218,10 +218,6 @@ def setLock(fileid, reqheaders, acctok):
                             (op.title(), fn, flask.request.args['access_token'][-20:], e))
 
     try:
-        # If LOCK and oldLock provided this is an UnLockAndRelock operation
-        if op == 'LOCK' and oldLock:
-            st.unlock(acctok['endpoint'], fn, acctok['userid'], acctok['appname'], utils.encodeLock(oldLock))
-
         # LOCK or REFRESH_LOCK: atomically set the lock to the given one, including the expiration time,
         # and return conflict response if the file was already locked
         st.setlock(acctok['endpoint'], fn, acctok['userid'], acctok['appname'], utils.encodeLock(lock))
```

Comment:
Why do we try a setlock when we already know that we are in a RefreshLock or UnLockAndRelock operation?

IMO this produces more requests than needed and pollutes the log.

Reply:
That's true, let me see how this can be streamlined. The original rationale was that `setLock` was atomic (whereas `refreshLock` was not), so it's the one used to protect from races.

Now that all those functions are expected to be atomic, we could indeed improve the logic.